banyan/src/store/mem_store.rs

```diff
@@ -43,12 +43,12 @@ impl<L: Eq + Hash + Copy> MemStore<L> {
         blocks.map.get(link).cloned()
     }
 
-    fn put0(&self, data: Vec<u8>) -> anyhow::Result<L> {
+    fn put0(&self, data: Vec<u8>) -> Result<L, Error> {
         let digest = (self.0.digest)(&data);
         let len = data.len();
         let mut blocks = self.0.blocks.lock();
         if blocks.current_size + data.len() > self.0.max_size {
-            anyhow::bail!("full");
+            return Err(Error::Full)
```

Comment:
… then this one could be `Error::MemStore("max_size would be exceeded")`

Reply:
You would consolidate the "multiple strong references" and the "max_size would be exceeded" error variants, as it seems. I'd rather keep them seperate, but make the error variants more expressive, to explain what happened...

See my patches :wink: